Add include directory to fix test that fails to find sleef.h during build on Intel systems when -march=native is set
Fix from https://github.com/pytorch/pytorch/issues/26555#issuecomment-579987873
Patch created by Simon Branford of the BEAR Software team at the University of Birmingham
--- caffe2/CMakeLists.txt.orig	2020-02-26 09:13:01.797943000 +0000
+++ caffe2/CMakeLists.txt	2020-02-26 09:13:40.870060513 +0000
@@ -1062,6 +1062,7 @@
     add_executable(${test_name} "${test_src}")
     target_link_libraries(${test_name} ${Caffe2_MAIN_LIBS} gtest_main)
     target_include_directories(${test_name} PRIVATE $<INSTALL_INTERFACE:include>)
+    target_include_directories(${test_name} PRIVATE $<BUILD_INTERFACE:${CMAKE_BINARY_DIR}/include>)
     target_include_directories(${test_name} PRIVATE ${Caffe2_CPU_INCLUDE})
     add_test(NAME ${test_name} COMMAND $<TARGET_FILE:${test_name}>)
     if (INSTALL_TEST)
